include/swift/AST/DiagnosticsSema.def

```diff
@@ -3264,6 +3264,18 @@ ERROR(autodiff_attr_result_not_differentiable,none,
       "'Differentiable', but %0 does not conform to 'Differentiable'", (Type))
 ERROR(autodiff_attr_opaque_result_type_unsupported,none,
       "cannot differentiate functions returning opaque result types", ())
+ERROR(autodiff_attr_static_decl_mismatch,none,
```

Comment:
Aren't these specific to the `@derivative` attribute? Should they be moved near `derivative_attr_*` diagnostics (around line 3171)?

Reply:
As I mentioned before the same diagnostic can apply to the `@transpose` attr. hence the definition in this section as the wording is not specific to `@derivative` nor `@transpose` attributes. Otherwise I would have to define them twice.

Reply:
Sounds good. The diagnostic name can use some clarification, something along the lines of `autodiff_attr_instance_method_mismatching_original`